File: backend/src/main/scala/microservices/tour-group/objects/CreateTourGroupPlanOptionPlannerRequest.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateTourGroupPlanOptionPlannerRequest(
    groupId: String,
    organizerUserId: String,
    planItemId: String,
    resourceType: String,
    resourceId: String,
    resourceVariantCode: Option[String] = None,
    resourceContext: Option[String] = None,
    label: String,
    description: String,
    defaultQuantity: Int
)
object CreateTourGroupPlanOptionPlannerRequest:
  given sourceEncoder: Encoder[CreateTourGroupPlanOptionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateTourGroupPlanOptionPlannerRequest] = deriveDecoder

File: frontend/src/microservices/tour-group/objects/CreateTourGroupPlanOptionPlannerRequest.ts
// This file defines `tour-group` module `CreateTourGroupPlanOptionPlannerRequest` as request payload data and provides JSON helpers.

export type CreateTourGroupPlanOptionPlannerRequest = {
  groupId: string
  organizerUserId: string
  planItemId: string
  resourceType: string
  resourceId: string
  resourceVariantCode: string | null
  resourceContext: string | null
  label: string
  description: string
  defaultQuantity: number
}
export const createTourGroupPlanOptionPlannerRequestFromJson = (json: string): CreateTourGroupPlanOptionPlannerRequest =>
  JSON.parse(json) as CreateTourGroupPlanOptionPlannerRequest

export const createTourGroupPlanOptionPlannerRequestToJson = (value: CreateTourGroupPlanOptionPlannerRequest): string =>
  JSON.stringify(value)
